dart-sample/build.gradle

```diff
@@ -1,49 +1,76 @@
+            'androidPlugin': '3.0.0',
+            'androidTools': '26.0.0',
+            'butterknife' : '8.8.1',
+            'release': '3.0.0',
+    ]
+
+    ext.deps = [
+            android: [
+                    'runtime': 'com.google.android:android:4.1.1.4',
+                    'gradlePlugin': "com.android.tools.build:gradle:${versions.androidPlugin}",
+                    'appCompat': "com.android.support:appcompat-v7:${versions.supportLibrary}"
+            ],
+            parceler: [
+                    'runtime': 'org.parceler:parceler-api:1.0.4',
+                    'compiler': 'org.parceler:parceler:1.0.4'
+            ],
+            dart: [
+                'runtime': "com.f2prateek.dart:dart:${versions.release}",
+                'compiler': "com.f2prateek.dart:dart-processor:${versions.release}"
+            ],
+            henson:[
+                'runtime': "com.f2prateek.dart:henson:${versions.release}",
+                'compiler': "com.f2prateek.dart:henson-processor:${versions.release}"
+            ],
+            butterknife:[
+                'runtime': "com.jakewharton:butterknife:${versions.butterknife}",
+                'compiler': "com.jakewharton:butterknife-compiler:${versions.butterknife}",
+                'annotations': "com.jakewharton:butterknife-annotations:${versions.butterknife}",
+                'gradlePlugin': "com.jakewharton:butterknife-gradle-plugin:${versions.butterknife}",
+                'gradlePluginSnapshot': 'com.jakewharton:butterknife-gradle-plugin:9.0.0-SNAPSHOT'
+            ],
+
+            junit: 'junit:junit:4.12',
+            fest: 'com.squareup:fest-android:1.0.7',
+            robolectric: 'org.robolectric:robolectric:2.4',
+    ]
 
-    defaultConfig {
-        applicationId 'com.f2prateek.dart.example'
-        minSdkVersion versions.minSdk
-        targetSdkVersion versions.compileSdk
-        versionCode 1
-        versionName '1.0.0'
+    repositories {
+        jcenter()
+        google()
+        maven {
+            url "https://plugins.gradle.org/m2/"
+        }
+        maven {
+            url "https://oss.sonatype.org/content/repositories/snapshots"
+        }
     }
 
-    lintOptions {
-        textReport true
-        textOutput 'stdout'
+    dependencies {
+        classpath deps.android.gradlePlugin
+        classpath deps.butterknife.gradlePlugin
+        classpath deps.butterknife.gradlePluginSnapshot
     }
 }
 
-dependencies {
-    implementation project(':dart-annotations')
-    implementation project(':dart')
-    implementation project(':henson')
-    implementation deps.parceler.runtime
-    implementation "com.jakewharton:butterknife:8.8.1"
-
-    annotationProcessor deps.parceler.compiler
-    annotationProcessor "com.jakewharton:butterknife-compiler:8.8.1"
-    annotationProcessor project(':dart-processor')
-    annotationProcessor project(':henson-processor')
-
-    testImplementation deps.junit
-}
+subprojects {
+    repositories {
```

Comment:
we can merge the dependencies of this root project and the sub projects by using `allprojects` instead of subprojects.